### PR TITLE
WIP: Add support for csi provider for azure

### PR DESCRIPTION
**Pull request by kartikjoshi21**
Not up for review yet. WIP PR

### CODE REVIEW

File: volumes/csi-wrapper/hack/azure/nginx-kata-with-my-pvc-and-csi-wrapper.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        - name: POD_UID
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.uid
+        - name: POD_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+      args:
+        - /usr/bin/csi-podvm-wrapper
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/nginx-kata-with-my-pvc-and-csi-wrapper.yaml:62]: maybe `- /usr/bin/csi-podvm-wrapper` is not need now.

File: volumes/csi-wrapper/hack/azure/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+1. Create the PeerpodVolume CRD object
+```bash
+cd /root/cloud-api-adaptor/
+kubectl create -f volumes/csi-wrapper/crd/peerpodvolume.yaml
+```
+The output looks like:
+```bash
+customresourcedefinition.apiextensions.k8s.io/peerpodvolumes.peerpod.azure.com created
+```
+2. Create vpc-block-csi-wrapper-runner role  bind to azurefiles-controller-sa account
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/README.md:50]: should `azure-files-csi-wrapper-runner` role here.

File: volumes/csi-wrapper/hack/azure/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+```bash
+kubectl create -f volumes/csi-wrapper/hack/hack/azure-files-csi-wrapper-runner.yaml
+```
+3. Build csi-wrapper images on worker node:
+```bash
+apt install docker.io -y
+cd volumes/csi-wrapper/
+make import-csi-controller-wrapper-docker
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/README.md:58]: if we have public `csi-controller-wrapper`, `csi-node-wrapper` and `csi-podvm-wrapper` docker images, this step can be skipped.
I have create a PR to build and push those csi-wrapper docker images, https://github.com/confidential-containers/cloud-api-adaptor/pull/1031

File: volumes/csi-wrapper/hack/azure/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+make import-csi-node-wrapper-docker
+make csi-podvm-wrapper-docker
+```
+4. Push csi-podvm-wrapper docker image to docker hub (Optional)
+```bash
+docker login -u [your_docker_hub_name] -p [your_docker_hub_password]
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/README.md:64]: if we have public `csi-controller-wrapper`, `csi-node-wrapper` and `csi-podvm-wrapper` docker images, this step can be skipped.
I have create a PR to build and push those csi-wrapper docker images, https://github.com/confidential-containers/cloud-api-adaptor/pull/1031

File: volumes/csi-wrapper/hack/azure/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+docker tag csi-podvm-wrapper:local [your_docker_hub_name]/csi-podvm-wrapper:dev
+docker push [your_docker_hub_name]/csi-podvm-wrapper:dev
+```
+5. patch ibm-vpc-block-csi-driver:
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/README.md:68]: should patch `csi-azurefile-driver`?

File: volumes/csi-wrapper/hack/azure/azure-file-StorageClass-for-peerpod.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: azure-file-storage
+provisioner: file.csi.azure.com
+parameters:
+  skuName: Standard_LRS
```
Comment (liudalibj) [inline volumes/csi-wrapper/hack/azure/azure-file-StorageClass-for-peerpod.yaml:7]: please add one new parameter
```
peerpod: "true"
```
It is need at https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/volumes/csi-wrapper/pkg/wrapper/controllerservice.go#LL41C2-L41C29

Comment (liudalibj) [reply volumes/csi-wrapper/hack/azure/azure-file-StorageClass-for-peerpod.yaml:7]: https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/volumes/csi-wrapper/pkg/wrapper/controllerservice.go#L85

**Comment by kartikjoshi21** [comment]
Closing this PR opened other one https://github.com/confidential-containers/cloud-api-adaptor/pull/1173

**Pull request closed without merging**